scripts/quests/windurst/To_Catch_A_Falling_Star.lua

```diff
@@ -0,0 +1,156 @@
+-----------------------------------
+-- To Catch A Falling Star
+-----------------------------------
+-- !addquest 2 53
+-- Sigismund : !pos -110 -9 80 206
+-----------------------------------
+require('scripts/globals/interaction/quest')
+require("scripts/globals/items")
+require("scripts/globals/npc_util")
+require('scripts/globals/quests')
+require('scripts/globals/zone')
+
+local westSaruMessages = require("scripts/zones/West_Sarutabaruta/IDs").text
+-----------------------------------
+local quest = Quest:new(xi.quest.log_id.WINDURST, xi.quest.id.windurst.TO_CATCH_A_FALLING_STAR)
+
+quest.reward =
+{
+    fame = 75,
+    fameArea = xi.quest.fame_area.WINDURST,
+    item = xi.items.FISH_SCALE_SHIELD
+}
+
+quest.sections =
+{
+    -- Quest Acceptance
+    {
+        check = function(player, status, vars)
+           return status == QUEST_AVAILABLE
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Sigismund'] = quest:progressEvent(196, 0, xi.items.STARFALL_TEAR),
+
+            onEventFinish =
+            {
+                [196] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    -- Supporting Character dialogue after Accepted
+    {
+        check  = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Tohopka'] = quest:progressEvent(198, 0, xi.items.STARFALL_TEAR, xi.items.HANDFUL_OF_PUGIL_SCALES)
+        }
+    },
+
+    -- Twinkle Tree in West Saru.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.WEST_SARUTABARUTA] =
+        {
+            ['Twinkle_Tree'] =
+            {
+                onTrigger = function(player, npc)
+                    if VanadielHour() <= 3 then
+                        player:messageSpecial(westSaruMessages.FROST_DEPOSIT_TWINKLES)
+                        player:messageSpecial(westSaruMessages.MELT_BARE_HANDS)
+                        return
+                    end
+                    player:messageSpecial(westSaruMessages.NOTHING_OUT_OF_ORDINARY)
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if VanadielHour() > 3 then
+                        player:messageSpecial(westSaruMessages.NOTHING_OUT_OF_ORDINARY)
+                        return
+                    end
+
+                    if npcUtil.tradeHas(trade, xi.items.HANDFUL_OF_PUGIL_SCALES) then
+                        player:messageSpecial(westSaruMessages.FROST_DEPOSIT_TWINKLES)
+                        player:messageSpecial(westSaruMessages.MELT_BARE_HANDS)
+                        if npcUtil.giveItem(player, xi.items.STARFALL_TEAR) then
+                            player:confirmTrade()
+                            if(quest:getVar(player, "Prog") == 0) then
```

Comment:
Quest framework styling uses single quotes around variables 
```suggestion
                            if quest:getVar(player, 'Prog') == 0 then
```
This will need to be changed elsewhere. Also note that if statements don't need brackets in lua

Reply:
Resolving all items in a single push in a few.

Reply:
see latest commit